OasysGH/Components/Helpers/Output.cs

```diff
@@ -7,28 +7,39 @@
 using UnitsNet.Serialization.JsonNet;
 using UnitsNet;
 using System;
+using Grasshopper.Kernel.Types;
 
 namespace OasysGH.Helpers
 {
   public class Output
   {
     private static UnitsNetIQuantityJsonConverter converter = new UnitsNetIQuantityJsonConverter();
-    public static void SetItem(GH_OasysDropDownComponent owner, IGH_DataAccess DA, int inputid, object data)
+    public static void SetItem<T>(GH_OasysDropDownComponent owner, IGH_DataAccess DA, int inputid, T data) where T : IGH_Goo
     {
       DA.SetData(inputid, data);
 
-      int outputsSerialized = 0;
+      string outputsSerialized = "";
       if (data.GetType() == typeof(GH_UnitNumber))
       {
-        IQuantity quantity = ((GH_UnitNumber)data).Value;
-        outputsSerialized = JsonConvert.SerializeObject(quantity, converter).GetHashCode();
+        IQuantity quantity = ((GH_UnitNumber)(object)data).Value;
+        outputsSerialized = JsonConvert.SerializeObject(quantity, converter);
       }
       else
-        outputsSerialized = JsonConvert.SerializeObject(data, converter).GetHashCode();
+      {
+        var obj = ((T)(object)data).ScriptVariable();
+        try
+        {
+          outputsSerialized = JsonConvert.SerializeObject(obj);
+        }
+        catch (Exception)
```

Comment:
sure you wanna catch any Exception here?

Reply:
Yes if JsonConvert doesn't work for any reason I don't want the component to fail. Preventing downstream objects from expiring is a nice feature, it's okay if it doesnt work...